### PR TITLE
PR: Add typing to schema_version

### LINKED ISSUE #2526: `schema_version` should be typed and preferably serialized to string
Everything in our settings models should have type annotations.  
Currently, all of the `schema_version` fields are untyped tuples of ints:
```python
class AppearanceSettings(BaseNapariSettings):
    schema_version = (0, 1, 0)
```
this means they serialize to our settings like this:
```yaml
appearance:
  schema_version:
  - 0
  - 1
  - 0
  theme: dark
```

which is a little hard to visually parse.  Since it's used in every submodel, it would be nice to have a `SchemaVersion` field that handles ser/des so that can be:
```yaml
appearance:
  schema_version: 0.1.0
  theme: dark
```

### PR DESCRIPTION
**Pull request by goanpeca**
# Description

Fixes #2526

@tlambert03 this takes a tuple and a string (to remain compatible with the old way of doing things)

Also added the check for typing: The test will fail like this:

image

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# References

# How has this been tested?

- [x] example: all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: napari/utils/settings/_defaults.py
Comment (sofroniewn) [inline napari/utils/settings/_defaults.py:17]: Can you add a doc string here

**Review by sofroniewn — APPROVED** [review]
Looks good to me (though can you add a quick doc string). I'll let @tlambert03 review before merge though

**Review by tlambert03 — APPROVED** [review]
excellent.  my clean yaml and I thank you

File: napari/utils/settings/_defaults.py
Comment (goanpeca) [reply napari/utils/settings/_defaults.py:17]: Fixed!

**Pull request merged**
